fabric-networking-api-v1/src/main/java/net/fabricmc/fabric/impl/networking/splitter/FabricPacketSplitter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.impl.networking.splitter;
+
+import java.util.List;
+import java.util.function.Consumer;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.MessageToMessageEncoder;
+
+import net.minecraft.network.handler.EncoderHandler;
+import net.minecraft.network.packet.Packet;
+import net.minecraft.network.packet.s2c.common.CustomPayloadS2CPacket;
+import net.minecraft.network.state.NetworkState;
+
+import net.fabricmc.fabric.mixin.networking.accessor.EncoderHandlerAccessor;
+
+public class FabricPacketSplitter extends MessageToMessageEncoder<Packet<?>> {
+	public static final int SAFE_SPLIT_SIZE = 1048576 - 2000;
```

Comment:
This needs a different value for serverbound. Plus you can use the constant:
```suggestion
	public static final int SAFE_SPLIT_SIZE = CustomPayloadS2CPacket.MAX_PAYLOAD_SIZE - 2000;
```

This value is also quite conservative. Wouldn't it be worth working out exactly what the maximum safe split size is? Currently it's `MAX_PAYLOAD_SIZE - 10`, 5 bytes each for the worst-case varint for `splitId` and `part`. However, if you implement my suggestion to remove the redundant data then this value should be equal to `MAX_PAYLOAD_SIZE`.

Reply:
I didn't do it as it was private. Through maybe I selected wrong source. Through yeah it could be lowered.